reframe/core/schedulers/__init__.py

```diff
@@ -243,11 +245,15 @@ def prepare(self, commands, environs=None, **gen_opts):
     def emit_preamble(self):
         pass
 
-    def guess_num_tasks(self):
+    def guess_num_tasks(self, min_num_tasks=1):
         if isinstance(self.sched_flex_alloc_tasks, int):
             if self.sched_flex_alloc_tasks <= 0:
                 raise JobError('invalid number of flex_alloc_tasks: %s' %
                                self.sched_flex_alloc_tasks)
+            elif self.sched_flex_alloc_tasks < min_num_tasks:
+                raise JobError('invalid number of flex_alloc_tasks: %s > '
+                               '%s (min number of tasks)' %
+                               self.sched_flex_alloc_tasks, min_num_tasks)
```

Comment:
The logic of the minimum tasks should not go inside the `guess_num_tasks()`. `guess_num_tasks()` should return the number of tasks that it has found; that's it. As before. Inside the prepare, you compare against the minimum and either return or raise a `JobError`. Going even further, the only reason I see raising a `JobError` from the `guess_num_tasks()` currently is just for having a more precise error message. You could even replace the raising with `return 0` and just log the message. In this case, the implementation will become more consistent.